app/assets/javascripts/actions/server_actions.js

```diff
@@ -6,9 +6,9 @@ const Flux = new McFly();
 const ServerActions = Flux.createActions({
 
   // General-purpose
-  fetch(model, courseId) {
+  fetch(model, courseId, args) {
```

Comment:
Rather than modify this widely-reused general 'fetch' function, we should follow the example of what we did for course revisions, with `fetchRevisions` in `revisions_actions.js`. This might be the right time to convert ArticleStore to redux, which would let you follow the revisions example pretty closely. But even if we're sticking with Flux and ArticleStore for the moment, we can still make a new fetchArticles action in articleActions to replace the use of API.fetch.

Reply:
@ragesoss since, the argument is directly passed as data and is the third argument, I thought it wouldn't do any harm. But yes, I'll change that.